src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/OpenAiEmbeddingClient.java

```diff
@@ -1,37 +1,39 @@
 package io.stargate.sgv2.jsonapi.service.embedding.operation;
 
 import io.quarkus.rest.client.reactive.QuarkusRestClientBuilder;
+import io.smallrye.mutiny.Uni;
 import jakarta.ws.rs.HeaderParam;
 import jakarta.ws.rs.POST;
 import jakarta.ws.rs.Path;
 import java.net.URI;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 import org.eclipse.microprofile.rest.client.annotation.ClientHeaderParam;
 import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
 
-public class OpenAiEmbeddingClient implements EmbeddingService {
+public class OpenAiEmbeddingClient implements EmbeddingProvider {
```

Comment:
Similar minor nitpick, the company name is "OpenAI" with an uppercase "I", so I'd prefer the EmbeddingClient name to match the company name exactly unless there's a good reason not to. And you already have "VertexAI" so there's precedent for uppercase "I"